src/orquestra/sdk/_base/_dsl.py

```diff
@@ -73,7 +73,12 @@ class UnknownPlaceholderInCustomNameWarning(Warning):
 # ----- data structures -----
 
 Constant = Any
-Argument = Union[Constant, "ArtifactFuture"]
+Argument = Union[Constant, "ArtifactFuture", "Secret"]
+
+
+class Secret(NamedTuple):
+    name: str
+    config_name: Optional[str] = None
```

Comment:
**non-blocking**: please add a comment describing when we expect this field to be `None`

Reply:
wdyt? 543cc88

Reply:
Nice!